File: LV3/3.0/4thought/4thought.py
op = ['+', '-', '*', '//']
combi_op = [(i, j, k) for i in op for j in op for k in op]
ans = {}


def gen_ans():
    for line in combi_op:
        eq = f"4 {' 4 '.join(line)} 4"
        val = eval(eq)
        ans[val] = f"{eq.replace('//', '/')} = {val}"


gen_ans()
n = int(input())

for _ in range(n):
    m = int(input())
    if m in ans:
        print(ans[m])
    else:
        print("no solution")

File: LV3/3.0/4thought/4thought.rb
def gen_ans
  ans = {}
  op = ['+', '-', '*', '/']

  op.product(op, op).each do |line|
    eq = "4 #{line.join(' 4 ')} 4"
    val = eval(eq)
    ans[val] = "#{eq} = #{val}"
  end
  ans
end

ans = gen_ans
n = gets.to_i

n.times do
  m = gets.to_i
  if ans.include?(m)
    puts ans[m]
  else
    puts 'no solution'
  end
end
